Fix encoding of post body.

RillingDev opened the pull request:
Hi! Thanks for the great library.
I was having issues submitting tags the other day, the tag "kayōkyoku" was always sent as "kay?kyoku".
I did some investigation and found that while the content type header for this request is set to UTF-8, the body uses the default encoding in the StringEntity constructor, which is ISO-8859-1.
This patch changes the encoding of the body to be UTF-8.

schnatterer commented on src/org/musicbrainz/webservice/impl/HttpClientWebServiceWs2.java:
```diff
@@ -1,12 +1,6 @@
-import org.apache.http.HttpResponse;
-import org.apache.http.HttpStatus;
-import org.apache.http.NoHttpResponseException;
+import org.apache.http.*;
```
Let's keep the changes as minimal as possible. So, please reconfigure your IDE to not touch the imports.

If you're happen to use IDEA, [here's how](https://stackoverflow.com/a/3348855/).

RillingDev replied:
Done, thanks for the link!

schnatterer commented on src/org/musicbrainz/webservice/impl/HttpClientWebServiceWs2.java:
```diff
@@ -185,19 +179,12 @@ protected Metadata doPost(String url, Metadata md) throws WebServiceException, M
 
-            StringEntity httpentity = new StringEntity(getWriter().getXmlString(md));
-            httpentity.setContentType(new BasicHeader("Content-Type", "application/xml; charset=UTF-8"));
+        StringEntity httpentity = new StringEntity(getWriter().getXmlString(md), Consts.UTF_8);
```
Same here: Let's keep the changes to a bare minimum. While I agree that the error handling here might not be the best idea, changing it might cause problems we're not aware of and break backward compatibility.

Or did the error handling cause problems for you?

RillingDev replied:
From what I see the catch block was only there because the default constructor was able to throw the checked exception UnsupportedEncodingException which is not the case if you pass the encoding directly. Using the new constructor no IOExceptions can be thrown from the block wrapped by try.

schnatterer reviewed (changes requested):
Hi @FelixRilling, 

I don't dare to say: You're welcome, because most of the work was actually done by @marcoc1712. 
Ages ago, when he was hosting it at Google Code, I forked musicbrainzws2-java to GitHub (which was the new kid in town back then) in order to do a couple of fixes myself. I always intended to bring both libraries together. Unfortunately, at the same time I did my fixes, Marco continued working on the original library. Somehow, I never managed to muster the time for the merge. So the fork persists :grimacing: 

Enough of history. Thanks for your PR! Interesting, that this lib is still in use after all those years.

I had a quick glance at the code but apart from that I can't judge if this works.
There are no unit tests and I honestly can't event try if this works. I only ever sent GET requests to MB :see_no_evil: 

Can you tell me how I can find out if this works?

RillingDev commented:
Thank you for the very fast response!

> I had a quick glance at the code but apart from that I can't judge if this works.
> There are no unit tests and I honestly can't event try if this works. I only ever sent GET requests to MB see_no_evil
> 
> Can you tell me how I can find out if this works?

Sure! Here is a simple snippet which fails in the current version that works after the patch:
```java
        ReleaseGroup releaseGroup = new ReleaseGroup();
        ReleaseGroupIncludesWs2 includes = new ReleaseGroupIncludesWs2();
        includes.includeAll();
        releaseGroup.setIncludes(includes);
        releaseGroup.getQueryWs().setClient("TestyMcTestface/0.0.1");
        releaseGroup.getQueryWs().setUsername("username");
        releaseGroup.getQueryWs().setPassword("password");

        releaseGroup.lookUp("mbid");

        releaseGroup.AddTags(new String[]{"kayōkyoku"});
```

schnatterer commented:
I implemented a way of running the existing integration test (it's called `UnitTest` :see_no_evil: ) against `test.musicbrainz.org`.
Could you refactor the snippet you wrote in your comment into a test? That way we can prohibit regressions. To do so, please rebase the upstream branch on yours.

Downside is that you'll probably need an account at `test.musicbrainz.org` for development.
In future this might be improved by using a containerized local MusicBrainz instance. I just don't have the time right now.

After you done, please squash to one commit. Then I'll release a new version. It's going to be available on Maven Central BTW :)

RillingDev committed:
Set encoding of post content to UTF-8 to match the charset already set in the content type.

RillingDev commented:
Added test and rebased into one commit.
1) I added a test without assertions (just using stdouts like the other tests), is that ok?
2) Travis build failed, but I *think* that is not due to my new test

schnatterer committed:
Make tests more reliable.

- Add assertion to encoding test. Otherwise it would not find the regression it was written for.
- Make UnitTests.java runnable without env vars.

schnatterer commented:
Thanks for the test!

1. I added an assertion. Otherwise the test would not fail with `kay?kyoku`, which was the reason for the whole thing.
2. Yes, travis failure was due to the credentials for the test server being passed in by travis via env vars, which is disabled for PRs. Luckily, I found out that all passwords on the test server are the same :grimacing: (well, good for us)

schnatterer commented:
Thanks again.  New version now released to maven central (you can get rid of the repo in `pom.xml`)

```
<dependency>
    <groupId>info.schnatterer.musicbrainzws2-java</groupId>
    <artifactId>musicbrainzws2-java</artifactId>
    <version>3.0.2</version>
</dependency>
```

RillingDev commented:
Thank you!
